File: backend/app/models/section.py
from sqlalchemy import Column, Integer, String, ForeignKey
from sqlalchemy.orm import relationship

from app.database.base_class import Base


class Section(Base):
    __tablename__ = "sections"

    id = Column(Integer, primary_key=True, index=True)

    property_id = Column(
        Integer,
        ForeignKey("properties.id"),
        nullable=False,
    )

    name = Column(String(100), nullable=False)

    description = Column(String(255), nullable=True)

    property = relationship("Property", back_populates="sections")

    units = relationship(
    "Unit",
    back_populates="section",
    cascade="all, delete",
)

File: frontend/safecolony_app/lib/features/admin/section/models/section.dart
class Section {
  final int id;
  final int propertyId;
  final String name;
  final String description;

  Section({
    required this.id,
    required this.propertyId,
    required this.name,
    required this.description,
  });

  factory Section.fromJson(Map<String, dynamic> json) {
    return Section(
      id: json["id"],
      propertyId: json["property_id"],
      name: json["name"],
      description: json["description"] ?? "",
    );
  }

  Map<String, dynamic> toJson() {
    return {
      "id": id,
      "property_id": propertyId,
      "name": name,
      "description": description,
    };
  }
}